### DISCUSSION TITLE
zRam on MicroOS

### DISCUSSION THREAD

**Discussion by aleksasiriski** [mysticaltech/terraform-hcloud-kube-hetzner#530] [Ideas]
Should we add an option to enable [zRam](https://software.opensuse.org/package/systemd-zram-service) and perhaps even make it the default?

**Comment by otavio** [comment]
I believe it so, I came from embedded systems background and where we use it a lot. 
 In your specific case, it's even improves performance. Especially because it increases the use of cache.

**Reply by aleksasiriski** [reply]
So it isn't enabled currently? I'll try to make a PR (it should just be a systemd service)

**Reply by mysticaltech** [reply]
@aleksasiriski Just SSH into a node and have a look! Usually, MicroOS is quite optimized as is.

![ksnip_20230125-133545](https://user-images.githubusercontent.com/518555/214564871-602d5361-0e1e-44db-bcb9-65e5e58735dc.png)

**Reply by mysticaltech** [reply]
@aleksasiriski I ran the above queries, and zRam is not active by default. It would be great to have it, but optional because of the reason highlighted below.

![ksnip_20230126-004927](https://user-images.githubusercontent.com/518555/214719700-1fc7da99-2cda-4fc4-9bda-e78e80bc12a9.png)

**Reply by ifeulner** [reply]
What is the use case? Adding additional features in all areas which are rarely used decreases the overall stability a lot (as we see already...)

**Reply by aleksasiriski** [reply]
If your services depend more on RAM than on CPU, which most of mine do, then zRam is essentially free ram (uses CPU to compress ram, and that acts as swap which recently got support by Kubernetes)

**Reply by mysticaltech** [reply]
@ifeulner Is right, if we do this, we should at least enforce it at the nodepool level, so that if we do not want it anymore, we just scale that nodepool to a count of 0. So it probably calls for an options variable in the nodepool definition?! Let's explore how that would work, intellectually or in a drawing maybe.

**Reply by mysticaltech** [reply]
Good use case for zRam! https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/issues/748#issuecomment-1518623079

**Comment by mysticaltech** [comment]
Now supported.
